spring-boot-project/spring-boot/src/main/java/org/springframework/boot/BeanDefinitionLoader.java

```diff
@@ -148,7 +169,7 @@ private int load(Object source) {
 	}
 
 	private int load(Class<?> source) {
-		if (isGroovyPresent() && GroovyBeanDefinitionSource.class.isAssignableFrom(source)) {
+		if (isGroovyPresent() && GroovyBeanDefinitionSource.class.isAssignableFrom(source) && IS_XML_ENABLED) {
```

Comment:
I guess the check for `IS_XML_ENABLED` could be the first to avoid any class lookups inside `isGroovyPresent` here.

Reply:
Indeed, but based on @wilkinsona feedback, I am going to avoid mixing things between the XML and the Groovy readers, so this check will be removed.